python/eos/figure/item.py

```diff
@@ -0,0 +1,178 @@
+        pass
+
+    def prepare(self):
+        "Setup of the item such as computing data points of an observable"
+        pass
+
+    def draw(self, ax):
+        """
+        Draw the item using a set of axis
+
+        :param ax: Matplotlib axis object
+        :type ax: matplotlib.axes
+        """
+
+        self.prepare()
+        # Call to matplotlib method on ax here
+        pass
+
+
+class Observable(Item):
+    """Represents an observable as a function of a variable.
+    The variable can be either a kinematic variable or a parameter.
+
+
+    :param observable: EOS qualified name of an observable
+    :type observable: str
+    :param variable: Name of a kinematic variable or of a parameter
+    :type variable: str
+    :param x_values: Values to be used as sampling points
+    :type x_values: list of float numbers
+    :param fixed_kineamtics: Names and values of fixed kinematic variables
+    :type fixed_kineamtics: dict
+    :param fixed_parameters: Names and values of fixed parameters
+    :type fixed_parameters: dict
+    :param fixed_parameters_from_file: Path to file that contains names and values of fixed parameters in the YAML format
+    :type fixed_parameters_from_file: str
+    :param options: Names and values of options passed to the EOS observable
+    :type options: dict
+    :param label: Label used in the legend
+    :type label: str
+    :param \**kwargs: Additional keyword arguments to pass to `matplotlib.axes.Axes.plot`
+    """
+
+    def __init__(self,
+            observable,
+            variable:str,
+            x_values:list,
+            fixed_kinematics:dict=None,
+            fixed_parameters:dict=None,
+            fixed_parameters_from_file:str=None,
+            options:dict=None,
+            label:str=None,
+            **kwargs
+        ):
+        super().__init__()
+
+        obs_entry = eos.Observables._get_obs_entry(observable)
+        valid_kin_vars = [kv for kv in obs_entry.kinematic_variables()]
+        eos.info('Plotting EOS observable "{}"'.format(observable))
+
+        self.label = label
+
+
+        # Create kinematics
+        eos_kinematics = eos.Kinematics()
+        if fixed_kinematics:
+            for k, v in fixed_kinematics.items():
+                if k not in valid_kin_vars:
+                    raise ValueError("Kinematic quantity '" + k + "' does not " +
+                    "match known ones for observable '" + observable + "': " + valid_kin_vars.__repr__())
+                eos_kinematics.declare(k, v)
+
+        # Create parameters
+        eos_parameters = eos.Parameters.Defaults()
+        if fixed_parameters and fixed_parameters_from_file:
+            eos.warn('Overriding values read from \'parameters-from-file\' with explicit values in \'parameters\'')
+            raise RuntimeError("Providing 'fixed_parameters_from_file' is not yet implemented")
+
+        elif fixed_parameters_from_file:
```

Comment:
Could you copy what @mreboud had recently added to the old plotting framework in terms of ``parameters-from-mode`` when plotting an observable?

Reply:
Sure thing. But if we add non-trivial features to the bare bones now, it is also a good time to add unit tests. If @mreboud already has a test case at hand, maybe he wants to add it? 😄

Reply:
Suspending this for the time being.